encoding/ccf/encode.go

```diff
@@ -0,0 +1,1808 @@
+		return e.encodeInt(x)
+
+	case cadence.Int8:
+		return e.encodeInt8(x)
+
+	case cadence.Int16:
+		return e.encodeInt16(x)
+
+	case cadence.Int32:
+		return e.encodeInt32(x)
+
+	case cadence.Int64:
+		return e.encodeInt64(x)
+
+	case cadence.Int128:
+		return e.encodeInt128(x)
+
+	case cadence.Int256:
+		return e.encodeInt256(x)
+
+	case cadence.UInt:
+		return e.encodeUInt(x)
+
+	case cadence.UInt8:
+		return e.encodeUInt8(x)
+
+	case cadence.UInt16:
+		return e.encodeUInt16(x)
+
+	case cadence.UInt32:
+		return e.encodeUInt32(x)
+
+	case cadence.UInt64:
+		return e.encodeUInt64(x)
+
+	case cadence.UInt128:
+		return e.encodeUInt128(x)
+
+	case cadence.UInt256:
+		return e.encodeUInt256(x)
+
+	case cadence.Word8:
+		return e.encodeWord8(x)
+
+	case cadence.Word16:
+		return e.encodeWord16(x)
+
+	case cadence.Word32:
+		return e.encodeWord32(x)
+
+	case cadence.Word64:
+		return e.encodeWord64(x)
+
+	case cadence.Fix64:
+		return e.encodeFix64(x)
+
+	case cadence.UFix64:
+		return e.encodeUFix64(x)
+
+	case cadence.Array:
+		return e.encodeArray(x, tids)
+
+	case cadence.Dictionary:
+		return e.encodeDictionary(x, tids)
+
+	case cadence.Struct:
+		return e.encodeStruct(x, tids)
+
+	case cadence.Resource:
+		return e.encodeResource(x, tids)
+
+	case cadence.Event:
+		return e.encodeEvent(x, tids)
+
+	case cadence.Contract:
+		return e.encodeContract(x, tids)
+
+	case cadence.Path:
+		return e.encodePath(x)
+
+	case cadence.TypeValue:
+		// cadence.TypeValue is encoded as self-contained, without any
+		// reference to tids.  So tids isn't passed to encodeTypeValue().
+		//
+		// encodeTypeValue() receives a new ccfTypeIDByCadenceType to deduplicate
+		// composite type values within the same CCF type value encoding.
+		// For example, when a composite type appears more than once
+		// (recursive or repeated as nested type) within the same type value,
+		// it is only encoded once and is subsequently represented by its CCF ID.
+		// For type value encoding, CCF type ID is sequentially generated by
+		// traversal order.
+		return e.encodeTypeValue(x.StaticType, ccfTypeIDByCadenceType{})
+
+	case cadence.StorageCapability:
+		return e.encodeCapability(x)
+
+	case cadence.Enum:
+		return e.encodeEnum(x, tids)
+
+	case cadence.Function:
+		// cadence.Function is encoded as self-contained, without any
+		// reference to tids.  So tids isn't passed to encodeFunction().
+		//
+		// encodeFunction() receives a new ccfTypeIDByCadenceType to deduplicate
+		// composite type values within the same CCF function value encoding.
+		// For example, when a composite type appears more than once
+		// (recursive or repeated as nested type) within the same function value,
+		// it is only encoded once and is subsequently represented by its CCF ID.
+		// For function value encoding, CCF type ID is sequentially generated by
+		// traversal order of sorted parameters and return type.
+		return e.encodeFunction(x.FunctionType, ccfTypeIDByCadenceType{})
+
+	default:
```

Comment:
There was a new Cadence value (and a corresponding type) introduced recently called `Attachment`, which may need handling.
e.g:
 https://github.com/onflow/cadence/blob/2e801370afbd4cc24f1e56c613257b49e97608bf/values.go#L1650-L1653

Reply:
Corresponding type:
https://github.com/onflow/cadence/blob/2e801370afbd4cc24f1e56c613257b49e97608bf/types.go#L1221-L1227